### DISCUSSION TITLE
Doesn't discover plug

### DISCUSSION THREAD

**Discussion by orbited303** [plasticrake/homebridge-tplink-smarthome#284] [Q&A]
Dear all,

Not quite sure where to post this – hope this will do.

So, I bought a KP115 plug at a very attractive price point not bothering about it not supporting HomeKit: Homebridge to my rescue, or so I thought. Turns out I've now spent hours and hours trying to get it to work. With no luck. And yeah, I should say that I have other devices running fine on Homebridge, but this is my first from TP Link. Given the time I've put into this, I should have just gone for a more expensive HomeKit plug, I guess. But now there's so much sunk cost, that I'd be damned if I can't get it working. But I've reached the end of the line and need your help.

Plugin install went fine. Logs show that homebridge-tplink-smarthome is loaded. After that, nothing. My plug is not discovered. (Works as expected in the Tapo app, though.)

Things I've tried:

**1. Re-installed Homebridge.**

**2. Made sure Homebridge and the KP115 is on the same subnet.**

**3. Rebooted everything: The server process, the physical server, the router.**

**4. Manually assigning IP (to 10.0.1.150) and broadcast address (255.255.255.255.)**

**5. Installed smarthome-api and running a search (as per another thread in the forum):**
```
user@machine ~ % tplink-smarthome-api search
Searching...
startDiscovery({
  discoveryInterval: 2000,
  discoveryTimeout: 10000,
  breakoutChildren: true,
  broadcast: '255.255.255.255'
})
```

**6. Setting up a child-bridge. (I don't know... I just ran out of ideas!)**

Starting Homebridge gives the following log output concerning TplinkSmarthome:
```
[11/24/2022, 2:40:03 PM] [TplinkSmarthome] Initializing TplinkSmarthome platform...
[11/24/2022, 2:40:03 PM] [TplinkSmarthome] Initializing child bridge 0E:60:00:CD:5C:C2
[11/24/2022, 2:40:03 PM] Homebridge v1.5.1 (HAP v0.10.4) (Homebridge 396A) is running on port 51853.
[11/24/2022, 2:40:04 PM] [TplinkSmarthome] Launched child bridge with PID 18816
[11/24/2022, 2:40:04 PM] Registering platform 'homebridge-tplink-smarthome.TplinkSmarthome'
[11/24/2022, 2:40:04 PM] [TplinkSmarthome] Loaded homebridge-tplink-smarthome v7.0.1 child bridge successfully
[11/24/2022, 2:40:04 PM] Loaded 0 cached accessories from cachedAccessories.0E6000CD5CC2.
[11/24/2022, 2:40:04 PM] [TplinkSmarthome] homebridge-tplink-smarthome v7.0.1, node v19.1.0, homebridge v1.5.1, api v2.7
[11/24/2022, 2:40:04 PM] Homebridge v1.5.1 (HAP v0.10.4) (TplinkSmarthome) is running on port 43169.
```

Tapo reports my plug to be:
Hardware version: 1.0
Firmware: 1.1.2 Build 220930 Release.144500

So, what should I do? :)

**Comment by schlosrat** [comment]
Hi orbited303, did you ever resolve this issue?

I'm having a similar problem discovering HS200 switches. I've got 6, and they all work in the Kasa app. They all show up in the router, and they're all on my guest WiFI with other devices I can detect like Nest and a Rachio sprinkler. I'm running a later HB (1.6.0), and an earlier node.js (16.16.0) on a Synology NAS wired into the router. Recently, two of the switches finally populated, but I'm not sure what caused that and the other 4 are still AWOL.

**Comment by orbited303** [comment]
No, it’s still dead! And not much attention here either. I guess, next time I’ll just go for something that is HomeKit/Thread/Matter compatible from the get-go. I’m getting too old for these hacks! :)

**Reply by schlosrat** [reply]
I'm thinking the same thing. I've ordered some supposedly homekit compatible switches to try replacing the a Kasa ones. I'm still trying to get the Kasa switches to work, but if the new ones come in and work out of the box then I'm returning this Kasa Krap while the return window is still open!
